tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -115,15 +116,15 @@ static void ascs_test_suite_after(void *f)
 {
 	int err;
 
-	/* If any of these fails, it's a fatal error for any tests running afterwards */
-	err = bt_bap_unicast_server_unregister_cb(&mock_bap_unicast_server_cb);
-	zassert_true(err == 0 || err == -EALREADY, "Unexpected err response %d", err);
-
-	/* Sleep to trigger any pending state changes from unregister_cb */
-	k_sleep(K_SECONDS(1));
-
 	err = bt_bap_unicast_server_unregister();
-	zassert_equal(err, 0, "Unexpected err response %d", err);
+
+	/* In case of a failure due to pending state change, try again */
+	while (err != 0 && err != -EALREADY) {
+		/* Only -EBUSY indicate pending state change */
+		zassert_equal(err, -EBUSY, "unexpected err response %d", err);
+		k_sleep(K_MSEC(10));
+		err = bt_bap_unicast_server_unregister();
+	}
```

Comment:
This isn't great IMO. 
`bt_bap_unicast_server_unregister` should be a synchronous action that should flush any pending state changes. 

Is that possible to do? IIRC there is some way to flush the system workqueue. Of course if this API is called in the system  workqueue thread we can't do that, can we?